File: string/7-anagram.php
<?php

$handle = fopen("php://stdin", "r");
fscanf($handle, "%d", $T);

while ($T --) {
    fscanf($handle, "%s", $str);

    if (strlen($str) % 2 == 1) {
        echo "-1\n";
        continue;
    }

    $count = 0;
    $letter = [];
    $length = strlen($str);
    for ($i=0; $i < 26; $i++) {
        $letter[$i] = 0;
    }

    for ($i=0; $i < $length / 2; $i++) {
        $letter[ord($str[$i]) - ord('a')] ++;
    }

    for ($i=$length / 2; $i < $length; $i++) {
        $letter[ord($str[$i]) - ord('a')] --;
    }

    for ($i=0; $i < 26; $i++) {
        $count += abs($letter[$i]);
    }

    echo $count / 2 . PHP_EOL;
}

File: String/7-anagram.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

int main(int argc, char const *argv[])
{
    int T;
    cin >> T;

    while (T --) {
        int count = 0;
        int letter[26] = {0};
        string s;
        cin >> s;

        int length = s.length();

        if (length % 2 == 1) {
            cout << -1 << endl;
            continue;
        }

        for (int i = 0; i < length / 2; ++i)
        {
            letter[s[i] - 'a'] ++;
        }

        for (int i = length / 2; i < length; ++i)
        {
            letter[s[i] - 'a'] --;
        }

        for (int i = 0; i < 26; ++i)
        {
            count += abs(letter[i]);
        }

        cout << count / 2 << endl;
    }

    return 0;
}
